### PR TITLE
[Blog] dask

### PR DESCRIPTION
**Pull request by noahnovsak**
First blog about dask, focusing on changes made to OWTable.

### CODE REVIEW

File: content/blog/2022/2022-12-13-dask-table.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+{{% figure src="/blog_img/2022/2022-12-13_dask_window.png" %}}
+
+The first steps to fixing these unjust assumptions have already been implemented under the hood. We have created a new way to store data using dask. The main advantage here is that instead of forcing all the data into memory we can instead, keep it on our hard drive until actually required for processing. And even then it won't be read all at once, but in manageable chunks. While this is sounds great, tying it into Data Table directly poses more issues. Accessing data from the hard drive can of course be very slow.
+
+One important disticntion, as we move towards storing data on a hard drive, is that more than processing the data, the bottleneck tends to be _reading_ the data in the first place. So while the simplest method to present a table stored in memory is to read each cell, one by one, and disply it, we see a marked performance increase if we instead, read the data from our hard drive in _chunks_.
```
Comment (markotoplak) [inline content/blog/2022/2022-12-13-dask-table.md:26]: The last sentence seems confusing. I see it is correct, but I almost understood it wrongly.

**Comment by markotoplak** [comment]
Thanks! I like it a lot! Cool introduction and schematics. This is much better that I could ever imagine doing myself.

Because this work has not been merged into Orange yet, perhaps you could somewhere link to your branch? And then, when merged, we can do a small edit.

File: content/blog/2022/2022-12-13-dask-table.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+thumbImage = "/blog_img/2022/2022-12-13_dask_data.png"
+frontPageImage = "/blog_img/2022/2022-12-13_dask_data.png"
+blog = ["dask", "widgets"]
+shortExcerpt = "Preparing Orange to handle much more data."
+longExcerpt = "A note on using dask to deal with more and more data with Orange."
++++
+
+As great as Orange is at simplifying machine learning and bringing it closer to your [everyday regular normal guy](https://www.youtube.com/watch?v=5PsnxDQvQpw), it turns out, there are some major drawbacks that stem from it's very inception. This is because Orange is meant to be run on a laptop. A device with just a handful of CPU cores, and maybe a couple of gigabytes of RAM. Due to this ideology we have so far reasoned that nobody in their right mind would want to throw a very large dataset at it. So most of Orange's components assume they can fit all the data into memory at once, process it on a single core, and expect results fairly quickly.
+
+You might see an issue here. What happens if I give Orange more data than my poor old laptop can handle?
```
Comment (markotoplak) [inline content/blog/2022/2022-12-13-dask-table.md:16]: I kinda like leaving it up to the imagination, but perhaps you can also give a hint of what happens. Your call.

File: content/blog/2022/2022-12-13-dask-table.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+{{% figure src="/blog_img/2022/2022-12-13_dask_chunking.png" %}}
+
+This approach does have another drawback though. Previously each element took an equal amount of time to read and display. Now we can quickly display everything from the chunk, but when we scroll past the edge of the chunk and nead to read a new one we have to wait a little while. This causes an uneven stuttering when scrolling, which is very unpleasant.
+
+Luckily, there's a simple solution for this one too. If I want to achieve a smoother scroll I have to let the table continue scrolling past the edge of my chunk, even though I don't know what it's supposed to display yet. Instead temporarily showing some empty cells while I ask dask to read the next chunk for me in the background. Once dask is finished preparing my data I can refresh my table with the actual values.
+
+This aproach actually ends up working surprisingly well. Now I can (in theory) use Orange to display arbitrarily large datasets without unwanted crashes or obscenely long wait times. It was also amazingly simple to implement, due to how well dask integrates with existing numpy code.
```
Comment (zalagruden) [inline content/blog/2022/2022-12-13-dask-table.md:34]: What surprised you about the approach? What did you expect, and what then didn't happen?

File: content/blog/2022/2022-12-13-dask-table.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
++++
+author = "Noah Novšak"
+date = "2022-12-13"
+draft = false
+title = "Orange you going to ask about dask"
+type = "blog"
+thumbImage = "/blog_img/2022/2022-12-13_dask_data.png"
+frontPageImage = "/blog_img/2022/2022-12-13_dask_data.png"
+blog = ["dask", "widgets"]
+shortExcerpt = "Preparing Orange to handle much more data."
```
Comment (markotoplak) [inline content/blog/2022/2022-12-13-dask-table.md:10]: I'd like longer shortExcerpts (3 lines on the first page) and longExcepts (~5+ lines in the Blog list).

That reminds me, we should have some guidelines about this.

**Comment by markotoplak** [comment]
Also, as @mitjapotocin just notified me, all images need meaningful alt text.

File: content/blog/2022/2022-12-13-dask-table.md (hunk 1 above)
Comment (noahnovsak) [reply content/blog/2022/2022-12-13-dask-table.md]: Updated, hopefully this is clearer.

**Pull request merged**